codex/slots/proofs/prover.nim

```diff
@@ -0,0 +1,98 @@
+## Nim-Codex
+## Copyright (c) 2024 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+##
+
+import pkg/chronos
+import pkg/chronicles
+import pkg/circomcompat
+import pkg/poseidon2
+import pkg/questionable/results
+
+import pkg/libp2p/cid
+
+import ../../manifest
+import ../../merkletree
+import ../../stores
+import ../../market
+import ../../utils/poseidon2digest
+
+import ../builder
+import ../sampler
+
+import ./backends
+import ../types
+
+export backends
+
+type
+  AnyProof* = CircomProof
```

Comment:
Why prefixing with AnyXXX and not just naming them XXX?

Reply:
because these are generic types that will represent all the proofs that we can handle, something like:

```nim
AnyProof* = CircomProof or KZGProof or XYZProof
```